### PR TITLE
Fix run on big machine

### PR DESCRIPTION
**Pull request by hying-caritas**
Fix two issues to run ftq on big machines with many CPU cores.

### CODE REVIEW

**Comment by rminnich** [comment]
Please accept my apologies for the delay in seeing your PR. This is fantastic work. Can you tell me how big the machine was? I will work to get this merged.

File: ftqcore.c [hunk 1]
```diff
@@ -42,6 +42,8 @@ void *ftq_core(void *arg)
 	/* core # is thread # */
 	wireme(thread_num);
 
+	usleep(20000);
```
Comment (rminnich) [inline ftqcore.c:45]: Why was this needed?

Comment (rminnich) [reply ftqcore.c:45]: is it possibly a bug in our synchronization?

Comment (brho) [reply ftqcore.c:45]: There was a bug in our sync that this would have avoided (volatile on 'hounds'), which I fixed on the master branch.

Though one thing I've been seeing (on Akaros on one machine, at least) is a clear 100 Hz signal for 180 ms after starting, regardless of frequency.  It was so clear that I actually saw the signal manually.  In a file with all 700 counts, I had about 10 600s spaced 100 samples apart at the beginning of the file.

Note that I had to adjust for the time it takes to do the 1000 warmups.  (e.g., 10 KHz sampling had 80 msec of interference in the output, but the 1000 samples takes 100 ms, and this pattern was confirmed for 20 KHz and 7 Khz).

I haven't tracked down this interference yet.  I'm running with Turbo mode enabled and our Core 0 timer tick is 100 Hz, so there's stuff I can look into.

My initial hunch is that the interference is external (since it's 100 Hz regardless of sample freq), and perhaps it has something to do with bash running for a little before it sleeps waiting on the process.  And when it runs for a while, it prevents my core from going into full turbo mode.  Though 180 ms is a long time.  So we'll see.

It's possible that there is another issue, and the reason for the 1000 warmups was to hide this warmup time, and on my machines the time for 1000 samples (100 msec at the default sampling freq) wasn't enough.

Oh, and usleep() might cause more issues than it solves on Akaros.  If we do need to delay, we should use a udelay() (busy wait, no resched, etc).

Comment (brho) [reply ftqcore.c:45]: I tried removing the 1000 sample (100 msec) warmup, and I confirmed that the 100 Hz interference runs for about 180 msec on my particular machine/OS setup.  Also, it's not our core 0 timer tick, since I changed that one's freq and the 100 Hz signal didn't move.

I tried on another machine (both Haswells), and it didn't have the warmup effect.  AFAIK, SMM isn't involved (MSR 0x34 reports no change).  Turning off turbo mode (actually setting the pstate to the max non-turbo ratio) didn't help.  The only thing that affected it was sleeping at C1 instead of C2.  Not sure what mystery is going on there.

One option would be to make the warmup time a parameter to the function.  That way you could tune it per machine.

Comment (hying-caritas) [reply ftqcore.c:45]: With "volatile hounds" fixes, this should be not necessary.  I will prepare the new pull request.

**Comment by brho** [comment]
I cherry-picked the commit "Fix CPU binding on Linux" and pushed it.  I also added an option to specify the delay amount, replacing the warmup loops.  Seems to work fine on my machine that has the weird 180 ms worth of 100 Hz interference.

So I think that should do it for this patch set.
